packages/runtime/src/state/modules/__tests__/api-resources.test.ts

```diff
@@ -0,0 +1,303 @@
+      ] as [APIResource, string][]
+    ).reduce(
+      (state, [resource, locale]) =>
+        APIResources.reducer(
+          state,
+          Actions.apiResourceFulfilled(resource.__typename, resource.id, resource, locale),
+        ),
+      APIResources.getInitialState(),
+    )
+
+    const updatedState = (
+      [
+        [{ ...swatch, hue: 17 }, null],
+        [{ ...pagePathnameSlice, pathname: 'test/new-pathname' }, null],
+        [
+          {
+            ...localizedPagePathnameSlice_it_IT,
+            pathname: 'test/new-pathname',
+            localizedPathname: 'test/new-localized-pathname-it-IT',
+          },
+          'it-IT',
+        ],
+        [{ ...localizedGlobalElement_fr_FR, data: { prompt: 'hello world' } }, 'fr-FR'],
+      ] as [APIResource, string][]
+    ).reduce(
+      (state, [resource, locale]) =>
+        APIResources.reducer(state, Actions.changeApiResource(resource, locale)),
+      state,
+    )
+
+    expect(APIResources.getSerializedState(updatedState)).toMatchSnapshot()
+  })
+
+  test('`Actions.evictApiResource` correctly removes the resource from the store', () => {
+    const state = (
+      [
+        [swatch, null],
+        [pagePathnameSlice, null],
+        [localizedPagePathnameSlice_fr_FR, 'fr-FR'],
+        [localizedPagePathnameSlice_it_IT, 'it-IT'],
+        [globalElement, null],
+        [localizedGlobalElement_fr_FR, 'fr-FR'],
+        [localizedGlobalElement_it_IT, 'it-IT'],
+      ] as [APIResource, string][]
+    ).reduce(
+      (state, [resource, locale]) =>
+        APIResources.reducer(
+          state,
+          Actions.apiResourceFulfilled(resource.__typename, resource.id, resource, locale),
+        ),
+      APIResources.getInitialState(),
+    )
+
+    const updatedState = (
+      [
+        [`${APIResourceType.Swatch}:${swatch.id}`, null],
+        [`${APIResourceType.PagePathnameSlice}:${localizedPagePathnameSlice_fr_FR.id}`, 'fr-FR'],
+        [`${APIResourceType.LocalizedGlobalElement}:${localizedGlobalElement_it_IT.id}`, 'it-IT'],
+      ] as [string, string][]
+    ).reduce(
+      (state, [id, locale]) => APIResources.reducer(state, Actions.evictApiResource(id, locale)),
+      state,
+    )
+
+    expect(APIResources.getSerializedState(updatedState)).toMatchSnapshot()
+  })
+
+  describe('Only getters for localizable resources accept the locale argument', () => {
+    const state = APIResources.getInitialState()
+    describe.each([
+      APIResourceType.Swatch,
+      APIResourceType.File,
+      APIResourceType.Typography,
+      APIResourceType.GlobalElement,
+      APIResourceType.Table,
+      APIResourceType.Snippet,
+      APIResourceType.Page,
+      APIResourceType.Site,
+    ])('getter for %s', resourceType => {
+      test.each(['fr-FR', null, undefined])('fails to compile with locale %s', locale => {
+        //@ts-expect-error
+        expect(APIResources.getHasAPIResource(state, resourceType, '', locale)).toBe(false)
```

Comment:
@migueloller Following up on your comment earlier, I disallowed passing a locale for non-localizable resources. You can still NOT pass a locale when calling a method for a localizable resource because disallowing that, while possible, breaks down function composition downstream.

Reply:
Example error:

<img width="1277" alt="typography locale error" src="https://github.com/user-attachments/assets/a89e13c4-67b5-4fb6-8012-7bb4a88ad236">